lib/Doctrine/DBAL/Driver/Mysqli/MysqliConnection.php

```diff
@@ -63,6 +63,19 @@ public function __construct(array $params, $username, $password, array $driverOp
 
         $this->_conn = mysqli_init();
 
+        if (isset($driverOptions['ssl_key']) &&
+            isset($driverOptions['ssl_cert']) &&
+            isset($driverOptions['ssl_ca'])
```

Comment:
As you say unfortunately PHP and MySQL docs are not clear respect about what minimum of  parameters are mandatory about secure connections.

I think the common use is with key, cert and ca as mandatory parameter, but I've talk with various colleagues and we think is possible stablish a secure connection with only key and cert.

In conclusion mandatory fields must be key and cert.
@deeky666 Are you agree?